File: appbfs.cpp
#include <iostream>
#include "graph.h"
#include "bfs.h"

using namespace std;

int main()
{
  Graph g("tinyG.txt");
  BreadthFirstSearch bfs(g, "0");

  for (auto const &v : g.getVerts())
  {
    cout << v << ": ";
    if (bfs.hasPathTo(v))
    {
      for (auto const &w : bfs.pathTo(v))
        cout << w << " ";
    }
    cout << endl;
  }
}

File: AppBFS.java
public class AppBFS {

	public static void main(String[] args) {
		Graph g = new Graph("tinyG.txt");
		BFS bfs = new BFS(g, "0");
		for(String v: g.getVerts()) {
			if (bfs.hasPathTo(v)) {
				System.out.println("Caminho para " + v + ": " + bfs.pathTo(v)
					+ " - dist: "+bfs.distTo(v));
			} else {
				System.out.println("Sem caminho para " + v);
			}
		}
	}
}
